### PR TITLE
MAPREDUCE-7475. Fixed non-idempotent unit tests

### PR DESCRIPTION
**Pull request by kaiyaok2**
## Description of PR
This PR fixes unit tests that are not idempotent and fail upon repeated execution within the same JVM instance due to self-induced state pollution. The tests shall be fixed as unit tests shall be idempotent, and it shall be good to clean the state pollution so that potential newly introduced tests do not fail in the future due to the shared state polluted by these tests.

## Overview & Proposed Fix of all non-idempotent unit tests in the MapReduce Module
The following tests try to make the directory `TEST_ROOT_DIR` and write to it. The tests do not clean up (remove) the directory after execution. Therefore, in the second execution, `TEST_ROOT_DIR` would already exist and the exception `Could not create test dir` would be thrown. 

* org.apache.hadoop.mapred.TestOldCombinerGrouping.testCombiner
* org.apache.hadoop.mapreduce.TestNewCombinerGrouping.testCombiner

## Sample Failure Message (in the 2nd run of the test):
```
java.lang.RuntimeException: Could not create test dir: /home/kaiyaok2/NIOExperiments/github.com/apache/hadoop/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/build/test/data/e4ee268d-2946-43a2-93b5-f2b4ac647279
        at org.apache.hadoop.mapreduce.TestNewCombinerGrouping.testCombiner(TestNewCombinerGrouping.java:109)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
```
The fix is done by fully deleting the test directory after test execution.

-----------------------------------------------------------------------------------------------

The following two tests below do not reset `NotificationServlet.counter`, so repeated runs throw assertion failures due to accumulation. 

- org.apache.hadoop.mapred.TestClusterMRNotification#testMR
- org.apache.hadoop.mapred.TestLocalMRNotification#testMR

Fixed by resetting `NotificationServlet.counter` and `NotificationServlet.failureCounter` to 0 after test execution.

-----------------------------------------------------------------------------------------------

The following test does not remove the key `AMParams.ATTEMPT_STATE`, so repeated runs of the test will not be missing the attempt-state at all:

- org.apache.hadoop.mapreduce.v2.app.webapp.TestAppController.testAttempts

Fixed by removing `AMParams.ATTEMPT_STATE` at the end of the test.

-----------------------------------------------------------------------------------------------

The following test fully deletes `TEST_ROOT_DIR` after execution, so repeated runs will throw a`DiskErrorException`:

- org.apache.hadoop.mapred.TestMapTask#testShufflePermissions

Fixed by checking if `TEST_ROOT_DIR` exists before test execution. Make the directory if not.

-----------------------------------------------------------------------------------------------

The following test does not restore the static variable `statusUpdateTimes` after execution, so consecutive runs throws `AssertionError`:

- org.apache.hadoop.mapred.TestTaskProgressReporter#testTaskProgress

Fixed by resetting `statusUpdateTimes` to 0 before test execution

## How was this patch tested?
After the patch, rerunning the tests in the same JVM does not produce any exceptions.

### CODE REVIEW

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java
Comment (steveloughran) [inline hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java]: best to do the fully delete here, so mkdirs always works, even if test was stopped partway through

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java
Comment (steveloughran) [inline hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java]: this doesn't get invoked if the job failed, and as there's no cleanup before L109, no recovery.

**Review by steveloughran — CHANGES_REQUESTED** [review]
commented on one test, same issue with the other.
* safest to deleteFully() at the start of the test, just before attempting mkdirs()
* otherwise, try/finally...but that doesn't recover from a jvm exit/crash

**Comment by kaiyaok2** [comment]
@steveloughran Thanks for the suggestions! I've made the changes accordingly.

**Comment by kaiyaok2** [comment]
> commented on one test, same issue with the other.
> 
> * safest to deleteFully() at the start of the test, just before attempting mkdirs()
> * otherwise, try/finally...but that doesn't recover from a jvm exit/crash

@steveloughran Thanks for the suggestions! I've made the changes accordingly.

**Review by steveloughran — APPROVED** [review]
LGTM
+1

**Comment by steveloughran** [comment]
will merge once the build completes.

**Review by ayushtkn — COMMENTED** [review]
why don't we just cleanup the testDir in After? If you run this test a couple of times and see the test_root_dir
```
ayushsaxena@ayushsaxena hadoop % ls -l /Users/ayushsaxena/code/hadoop/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/target/test-dir/
total 0
drwxr-xr-x@ 4 ayushsaxena  staff  128 May  4 12:13 22d6c31d-6195-41d5-b3a0-835940271ce5
drwxr-xr-x@ 4 ayushsaxena  staff  128 May  4 12:12 6a754c2e-3c13-4b85-94fa-8a654ebdf936
drwxr-xr-x@ 4 ayushsaxena  staff  128 May  4 12:16 ab9a29a6-8210-476e-8912-e6d9fdcd40e3
drwxr-xr-x@ 4 ayushsaxena  staff  128 May  4 12:15 e22b7045-cefa-4979-b94c-1a5edb2eea73
ayushsaxena@ayushsaxena hadoop % 

```

it does have all those directories, MR tests don't delete the testDir in the pom either.

Why don't we try something like this
```
diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestOldCombinerGrouping.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestOldCombinerGrouping.java
index 046c2d37eed9..d6b4952f9d6a 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestOldCombinerGrouping.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestOldCombinerGrouping.java
@@ -18,11 +18,16 @@
 
 package org.apache.hadoop.mapred;
 
+import org.junit.After;
 import org.junit.Assert;
+
+import org.apache.hadoop.fs.FileUtil;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.io.LongWritable;
 import org.apache.hadoop.io.RawComparator;
 import org.apache.hadoop.io.Text;
+import org.apache.hadoop.test.GenericTestUtils;
+
 import org.junit.Test;
 
 import java.io.BufferedReader;
@@ -34,12 +39,9 @@
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Set;
-import java.util.UUID;
 
 public class TestOldCombinerGrouping {
-  private static String TEST_ROOT_DIR = new File(System.getProperty(
-      "test.build.data", "build/test/data"), UUID.randomUUID().toString())
-          .getAbsolutePath();
+  private static File TEST_ROOT_DIR = GenericTestUtils.getRandomizedTestDir();
 
   public static class Map implements
       Mapper<LongWritable, Text, Text, LongWritable> {
@@ -117,9 +119,14 @@ public int compare(Text o1, Text o2) {
 
   }
 
+  @After
+  public void cleanup() {
+    FileUtil.fullyDelete(TEST_ROOT_DIR);
+  }
+
   @Test
   public void testCombiner() throws Exception {
-    if (!new File(TEST_ROOT_DIR).mkdirs()) {
+    if (!TEST_ROOT_DIR.mkdirs()) {
       throw new RuntimeException("Could not create test dir: " + TEST_ROOT_DIR);
     }
     File in = new File(TEST_ROOT_DIR, "input");

```

 but I am kind of ok with the current impl as well if @steveloughran is convinced

**Comment by kaiyaok2** [comment]
@ayushtkn Thanks for the review! I agree that its better to cleanup the test directory in `@After`. I have addressed the changes you've suggested.

**Comment by slfan1989** [comment]
@kaiyaok2 Can #6790 and #6785 be fixed together? 
cc: @ayushtkn @steveloughran

**Comment by kaiyaok2** [comment]
@slfan1989 Yes. I merged all code changes in #6790 to this PR (#6785)

**Review by goiri — APPROVED** [review]

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 01s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 01s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 59s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  | 109m 16s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   8m 00s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   5m 43s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  17m 55s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  16m 52s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  | 205m 05s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   3m 04s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   9m 35s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 50s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 50s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m 00s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/1/artifact/out/blanks-eol.txt) |  The patch has 4 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  checkstyle  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   9m 30s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   8m 00s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 204m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   6m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 563m 16s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6785 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 1ac01a056e3d 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | trunk / 6e7c3ec42a278683c5d0b34242e1889ca8a67612 |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/1/testReport/ |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/1/console |
| versions | git=2.44.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by kaiyaok2** [comment]
Resolved the trailing space issues

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 01s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 00s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 20s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  85m 29s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   6m 18s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   4m 36s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  14m 16s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  13m 49s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  | 157m 54s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 03s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 03s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 00s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 21s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   7m 29s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   6m 28s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 160m 01s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   5m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   | 439m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6785 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 2bb0df688968 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | trunk / 4e98ad73e02de48c4078314ec240f4106faf81dc |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/2/testReport/ |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/2/console |
| versions | git=2.44.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
@ayushtkn the issue with @after is doesn't clean up reliably on job interrupt. but that's more of an issue with cloud stuff, isn't it? as here a `mvn clean` will do the cleanup.

So yes, let's go with your proposal

**Review by ayushtkn — APPROVED** [review]
yep, mvn clean should do the cleanup here.

The last few builds are from the windows precommit which doesn't run the tests, It got enabled again & it doesn't let the normal builds run this time I believe.

I have manually started the normal build for this to get the test results:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/

If the tests pass, will hit the merge button

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  43m 20s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 32s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  10m 50s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  compile  |   1m 34s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 55s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 31s |  |  trunk passed  |
| +1 :green_heart: |  spotbugs  |   3m  9s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  39m 12s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 30s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 33s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 33s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m  4s | [/results-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/artifact/out/results-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client.txt) |  hadoop-mapreduce-project/hadoop-mapreduce-client: The patch generated 2 new + 26 unchanged - 2 fixed = 28 total (was 28)  |
| +1 :green_heart: |  mvnsite  |   1m 36s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m  3s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  37m 52s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 43s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 57s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  | 120m 16s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 57s |  |  The patch does not generate ASF License warnings.  |
|  |   | 311m 19s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6785 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux cbd8f53faff6 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 4e98ad73e02de48c4078314ec240f4106faf81dc |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/testReport/ |
| Max. process+thread count | 1150 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/console |
| versions | git=2.9.5 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ayushtkn** [comment]
We have the test results, everything is green, the checkstyle alarm is for the existing code, not introduced as part of this PR but I think since we are touching it now, maybe we can fix that as well
@kaiyaok2 can you fix these checkstyle warnings
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6785/7/artifact/out/results-checkstyle-hadoop-mapreduce-project_hadoop-mapreduce-client.txt

it is just the variable name change, maybe changing to ``testRootDir`` should fix

**Comment by kaiyaok2** [comment]
> it is just the variable name change, maybe changing to `testRootDir` should fix

@ayushtkn Fixed. Thanks for the message

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 01s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 00s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 25s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  93m 25s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   7m 17s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   5m 01s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  15m 39s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  14m 42s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  | 176m 23s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   8m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 10s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 10s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 01s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 33s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   8m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   7m 06s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 178m 32s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   6m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 487m 18s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6785 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 285952dfb8f1 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | trunk / 710e1ce492e199eeb586bfd26eb81c95bddbcadf |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/3/testReport/ |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/3/console |
| versions | git=2.45.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 01s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 00s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 29s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  | 108m 49s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   7m 46s |  |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   5m 39s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |  17m 52s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |  16m 31s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  | 201m 25s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 44s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   9m 31s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 53s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 53s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 00s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   9m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   7m 58s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 201m 35s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   6m 25s |  |  The patch does not generate ASF License warnings.  |
|  |   | 556m 53s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6785 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 f036a21db0b9 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | trunk / e4e557d43c7575dc5a49821dd609edb1927790c4 |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/5/testReport/ |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6785/5/console |
| versions | git=2.45.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
thanks! merged to trunk. Can you do a cherrypick to branch-3.4 and push that up as a PR too so that yetus can validate the backport. Then we can merge it in there too

**Comment by kaiyaok2** [comment]
@steveloughran Done in https://github.com/apache/hadoop/pull/6837

Please lmk if I shall do the same for https://github.com/apache/hadoop/pull/6793

**Pull request merged**
